bots/run-queue

```diff
@@ -21,40 +21,73 @@ MAX_PRIORITY = 9
 
 import argparse
 import json
+import random
+import ssl
 import subprocess
 import sys
+import urllib.request, urllib.parse, urllib.error
+
+import pika
+from task import REDHAT_PING
+
+# Check if we have access to Red Hat network
+try:
+    urllib.request.urlopen(REDHAT_PING).read()
+    run_redhat_tasks = True
+except IOError:
+    run_redhat_tasks = False
 
-import amqp
 
 def main():
     parser = argparse.ArgumentParser(description='Bot: read a single test command from the queue and execute it')
-    parser.add_argument('--amqp', default='localhost',
-                        help='The URL of the AMQP server to consume from')
-    parser.add_argument('--queue', default='tasks',
-                        help='The name of the queue to consume from')
+    parser.add_argument('--amqp', default='localhost:5671',
+                        help='The URL of the AMQP server to consume from (default: %(default)s)')
     opts = parser.parse_args()
 
-    connection = amqp.Connection(host=opts.amqp)
-    connection.connect()
+    try:
+        host, port = opts.amqp.split(':')
+    except ValueError:
+        sys.stderr.write('Please format URL as host:port\n')
+        return 1
+
+    connection = pika.BlockingConnection(pika.ConnectionParameters(
+        host=host,
+        port=int(port),
+        ssl=True,
+        ssl_options=pika.SSLOptions(
+            ssl_version=ssl.PROTOCOL_TLSv1_2,
+            cafile='/run/secrets/webhook/ca.pem',
+            keyfile='/run/secrets/webhook/amqp-client.key',
+            certfile='/run/secrets/webhook/amqp-client.pem',
+            server_hostname=host),
+        credentials=pika.credentials.ExternalCredentials()))
     channel = connection.channel()
+
     try:
-        channel.queue_declare(queue=opts.queue, passive=True, auto_delete=False)
-    except amqp.exceptions.NotFound:
-        sys.stdout.write('queue is empty\n')
+        declare_public_result = channel.queue_declare(queue='public', passive=True, auto_delete=False)
+        declare_rhel_result = channel.queue_declare(queue='rhel', passive=True, auto_delete=False)
+    except pika.exceptions.ChannelClosed:
+        print('queue is empty')
         return 0
```

Comment:
This logic would terminate the program if *either* queue is empty. At most this should happen if *both* are empty?

Reply:
oh right, well it's more if either doesn't exist actually. Which won't happen because tests-scan creates them both at the same time. The error message is a bit bad.